api/internals/init.md

```diff
@@ -0,0 +1,169 @@
+# IPFS Node Init Specification
+
+> This is the specification for the steps performed when a new IPFS node is initialized.
+
+*Note that initializing an IPFS node differs from initializing an IPFS repo. For more on repo initialization, read the [Repo specification](https://github.com/ipfs/specs/tree/master/repo).*
+
+## Status
+
+First draft.
+
+For now, [`ipfs/go-ipfs/cmd/ipfs/init.go`](https://github.com/ipfs/go-ipfs/blob/master/cmd/ipfs/init.go) is the source of truth.
+
+## Flags
+
+There are some optional flags that may be supplied to `ipfs init`:
+
+```
+ -f|--force: whether to overwrite an existing repo config. Defaults to false.
+ -e|--empty-repo: whether to add+pin ipfs info help files to the repo. Defaults to false.
+ -b|--bits: # of bits to use in the generated RSA private key. Defaults to 2048.
+```
```

Comment:
there's an important difference between:
- an ipfs node (logical idea of what an ipfs node is)
- a commandline tool that inits an ipfs node (a projection of the abstract concept into a concrete implementation, in this case a cli tool based one)

the upshot is that we may need two docs:
- one doc that describes initializing an ipfs node, namely might specify "options", but not "commandline flags". for example, `IPFS_PATH` env var would NOT belong here. (may be too abstract for this doc).
- a doc that describes how initializing an ipfs node happens through cli tools like go-ipfs and js-ipfs (more this doc)